Classes_Homework_5/fromxml.py

```diff
@@ -0,0 +1,141 @@
+        r_text = self.get_element( my_sect, my_label, txt, strat)
+        r_cal = self.get_element( my_sect, my_label, cal, strat)
+        fit_tip = e.get_fitness_message(r_cal)
+        return r_text, r_cal, fit_tip
+
+    def process_xml(self, my_section, strat):
+        """Within each section format values for file + insert them to db"""
+        section_label = self.get_attr(my_section)
+        com = Combine(section_label, decor, decor_length)  # instantiate Combine class which puts together and decorates sections based on city/text/date etc attributes
+        if section_label == label1:  # label1 = "News"
+            n_city, n_text, n_date = self.news_values(my_section, section_label, strat)
+            return com.get_news(n_city, n_text, n_date)+'\n\n', [section_label, n_city, n_text, n_date]
+        elif section_label == label2:  # label2 = "Ad"
+            a_text, exp_date, a_date = self.ad_values(my_section, section_label, strat)
+            return com.get_ad(a_text, exp_date, a_date)+'\n\n', [section_label, a_text, a_date]
+        elif section_label == label3:  # label3 = "Recipe"
+            r_text, r_cal, fit_tip = self.recipe_values(my_section, section_label, strat)
+            return com.get_recipe(r_text, r_cal, fit_tip)+'\n\n', [section_label, r_text, r_cal, fit_tip]
+
+    def verify_source(self, my_path):
+        """Try opening and parsing the file, if any errors - raise exceptions"""
+        try:
+            sections = self.get_sections(my_path)
+            strategy = self.choose_strategy()
+            text_values = []
+            db_values = []
+            for section in sections:
+                processed_text, values = self.process_xml(section, strategy)
+                if processed_text:
+                    text_values.append(processed_text)
+                    db_values.append(values)
+            self.raise_if_empty(text_values)  # if list is empty, raise error
+            final_text = ''.join(text_values).rstrip()  # join list of formatted sections back into text
+            return my_path, final_text, db_values
+        except (OSError, ValueError, ET.ParseError, NoSectionsError,
+                PastDate, NoValue) as e:
+            print(e)
+            return None, None, None
+        except AttributeError:
+            print(cnf.get_values("ERRORS", "cannot_parse"))
+            return None, None, None
+        except KeyError:
+            print(cnf.get_values("ERRORS", "no_attr")+"\n")
+            return None, None, None
+        except TypeError:
+            pass
+            return None, None, None
+
+
+if __name__ == "__main__":
+    default_source = cnf.get_values("PATHS", "xml_source")
+    default_target = cnf.get_values("PATHS", "target_file")
+    fx = FromXML(default_source, default_target)
+    raw_source = fx.parse()  # parse method is inherited from FromJson
+    fx.file_full_flow(raw_source)  # method inherited from FromJson->WriteFromFile
+    cnt.write_csv()
```

Comment:
I was slightly confused to see "write_**csv**" method in the file which makes **xml** processing. This underscores how important (and sometimes not easy) is to create a correct method name :)